eras/conway/impl/src/Cardano/Ledger/Conway/TxInfo.hs

```diff
@@ -199,20 +205,62 @@ instance
     PlutusPurposeNotSupported purpose ->
       kindObject "PlutusPurposeNotSupported" ["purpose" .= toJSON purpose]
 
+-- | Given a TxOut, translate it for V2 and return (Right transalation).
+-- If the transaction contains any Byron addresses or Babbage features, return Left.
+conwayTransTxOutV1 ::
+  forall era.
+  ( Inject (BabbageContextError era) (ContextError era)
+  , Value era ~ MaryValue (EraCrypto era)
+  , BabbageEraTxOut era
+  ) =>
+  TxOutSource (EraCrypto era) ->
+  TxOut era ->
+  Either (ContextError era) PV1.TxOut
+conwayTransTxOutV1 txOutSource txOut = do
+  when (isSJust (txOut ^. dataTxOutL)) $ do
+    Left $ inject $ InlineDatumsNotSupported @era txOutSource
+  case Alonzo.transTxOut txOut of
+    Nothing -> Left $ inject $ ByronTxOutInContext @era txOutSource
+    Just plutusTxOut -> Right plutusTxOut
+
+-- | Given a TxIn, look it up in the UTxO. If it exists, translate it to the V1 context
+conwayTransTxInInfoV1 ::
+  forall era.
+  ( Inject (BabbageContextError era) (ContextError era)
+  , Value era ~ MaryValue (EraCrypto era)
+  , BabbageEraTxOut era
+  ) =>
+  UTxO era ->
+  TxIn (EraCrypto era) ->
+  Either (ContextError era) PV1.TxInInfo
+conwayTransTxInInfoV1 utxo txIn = do
+  txOut <- left (inject . AlonzoContextError @era) $ Alonzo.transLookupTxOut utxo txIn
+  plutusTxOut <- conwayTransTxOutV1 (TxOutFromInput txIn) txOut
+  Right (PV1.TxInInfo (transTxIn txIn) plutusTxOut)
+
 instance Crypto c => EraPlutusTxInfo 'PlutusV1 (ConwayEra c) where
   toPlutusTxCert _ = transTxCertV1V2
 
   toPlutusScriptPurpose proxy = transPlutusPurposeV1V2 proxy . hoistPlutusPurpose toAsItem
 
   toPlutusTxInfo proxy pp epochInfo systemStart utxo tx = do
-    let refInputs = txBody ^. referenceInputsTxBodyL
-    unless (Set.null refInputs) $ Left $ inject $ ReferenceInputsNotSupported @(ConwayEra c) refInputs
-
     timeRange <- Alonzo.transValidityInterval pp epochInfo systemStart (txBody ^. vldtTxBodyL)
-    inputs <- mapM (Babbage.transTxInInfoV1 utxo) (Set.toList (txBody ^. inputsTxBodyL))
+    inputs <- mapM (conwayTransTxInInfoV1 utxo) (Set.toList (txBody ^. inputsTxBodyL))
+    let
+      hasInlineDatum txIn =
+        case txinLookup txIn utxo of
+          Just txOut
+            | isSJust (txOut ^. dataTxOutL) -> True
+          _ -> False
+      refInlineDatum = find hasInlineDatum . toList $ txBody ^. referenceInputsTxBodyL
+    case refInlineDatum of
+      Just txIn ->
+        Left . BabbageContextError . InlineDatumsNotSupported $
+          TxOutFromInput txIn
+      Nothing -> pure ()
```

Comment:
You need to use `conwayTransTxInInfoV1` translation here on ref inputs while discarding the results, because there are other checks besides the inline datums